src/NuGet.Core/NuGet.Packaging/PackagesConfigWriter.cs

```diff
@@ -524,10 +525,7 @@ public void WriteFile(string fullPath)
                     @"packages.config.old." + DateTime.Now.ToString("yyyyMMddHHmmss"));
 
                 // Delete configFileCopyPath if it already exists
-                if (File.Exists(configFileCopyPath))
```

Comment:
``FileUtility.Delete`` does a ``File.Exists``   The check is primarily to avoid DirectoryNotFound exceptions. File.Delete will not throw if the file is missing.